g2o/core/base_constant_edge.h

```diff
@@ -0,0 +1,185 @@
+    using NthType = typename std::tuple_element<N, std::tuple<Types...>>::type;
+
+    template<typename Value, typename... Ts2>
+    auto tuple_init(const Value& value, const std::tuple<Ts2...>)
+    {
+      return std::tuple<Ts2...>{Ts2{value}...};
+    }
+
+    template<int EdgeDimension, int VertexDimension>
+    using JacobianType = typename Eigen::Matrix<number_t, EdgeDimension, VertexDimension, EdgeDimension==1?Eigen::RowMajor:Eigen::ColMajor>::AlignedMapType;
+  }
+
+  template <int D, typename E, typename... VertexTypes>
+  class BaseConstantEdge : public BaseEdge<D, E>
+  {
+    public:
+      template<int VertexN>
+      using VertexXnType = NthType<VertexN, VertexTypes...>;
+      template<int VertexN>
+      static constexpr int VertexDimension() { return VertexXnType<VertexN>::Dimension; };
+      template<int VertexN>
+      const auto vertexXn() const
+      {
+        return static_cast<const VertexXnType<VertexN>*>(_vertices[VertexN]);
+      }
+      template<int VertexN>
+      auto vertexXn()
+      {
+        return static_cast<VertexXnType<VertexN>*>(_vertices[VertexN]);
+      }
+
+      static const int Dimension = BaseEdge<D, E>::Dimension;
+      typedef typename BaseEdge<D,E>::Measurement Measurement;
+
+      typedef typename BaseEdge<D,E>::ErrorVector ErrorVector;
+      typedef typename BaseEdge<D,E>::InformationType InformationType;
+
+      //! it requires quite some ugly code to get the type of hessians...
+      template<int DN, int DM>
+      using HessianBlockType = Eigen::Map<Eigen::Matrix<number_t, DN, DM, DN==1?Eigen::RowMajor:Eigen::ColMajor>,
+                                          Eigen::Matrix<number_t, DN, DM, DN==1?Eigen::RowMajor:Eigen::ColMajor>::Flags & Eigen::PacketAccessBit ? Eigen::Aligned : Eigen::Unaligned >;
+      template<int K>
+      using HessianBlockTypeK = HessianBlockType<VertexDimension<index_to_pair(K).first>(), VertexDimension<index_to_pair(K).second>()>;
+      template<int K>
+      using HessianBlockTypeKTransposed = HessianBlockType<VertexDimension<index_to_pair(K).second>(), VertexDimension<index_to_pair(K).first>()>;
+      template<typename> struct HessianTupleType;
+      template<std::size_t... Ints >
+      struct HessianTupleType<std::integer_sequence<std::size_t, Ints...>>
+      {
+        using type = std::tuple<HessianBlockTypeK<Ints>...>;
+        using typeTransposed = std::tuple<HessianBlockTypeKTransposed<Ints>...>;
+      };
+      static const auto _nr_of_vertices = sizeof...(VertexTypes);
+      static const auto _nr_of_vertex_pairs = pair_to_index(0, _nr_of_vertices);
+      using HessianTuple = typename HessianTupleType<std::make_index_sequence<_nr_of_vertex_pairs>>::type;
+      using HessianTupleTransposed = typename HessianTupleType<std::make_index_sequence<_nr_of_vertex_pairs>>::typeTransposed;
+
+      BaseConstantEdge() : BaseEdge<D,E>(),
+      _hessianRowMajor(false),
+      _hessianTuple(tuple_init(nullptr, _hessianTuple)),
```

Comment:
your tuple_init is equivalent to as your just trying to null them out:

_hessianTuple{},
_hessianTupleTransposes{}

Reply:
(as you round out above, this is not possible because of Eigen's choice not to have them default constructable.)